### PR TITLE
Update to cardano-api 8.38

### PR DESCRIPTION
**Pull request by sjoerdvisscher**

### CODE REVIEW

**Review by MaximilianAlgehed — APPROVED** [review]

**Comment by MaximilianAlgehed** [comment]
Actually, is there a way to maintain backwards compatibility? The minor version bump in `cardano-api` suggests we shoudn't have to tighten things so much/

File: quickcheck-contractmodel/quickcheck-contractmodel.cabal [hunk 1]
```diff
@@ -89,6 +89,6 @@ library
       quickcheck-threatmodel
     -- Plutus and Cardano dependencies
     build-depends:
-      cardano-api                                     >= 8.36 && < 8.37,
-      cardano-ledger-core                             >= 1.2.0 && < 1.10,
-      cardano-ledger-shelley                          >= 1.2.0 && < 1.9,
+      cardano-api                                     >= 8.38 && < 8.39,
```
Comment (MaximilianAlgehed) [inline quickcheck-contractmodel/quickcheck-contractmodel.cabal:92]: This is a little bit scary, we are trying to use `ThreatModel` with `hydra` - which depends on `8.37`. What's the versioning scheme of `cardano-api`?

**Comment by sjoerdvisscher** [comment]
`cardano-api` is breaking backwards compatibility all the time. You'll get a lot of CPP if you want to keep backwards compatibility.

Are you sure Hydra will stay at 8.37?

**Comment by MaximilianAlgehed** [comment]
I don't know what hydra will do, but I'm now slightly upset over `cardano-api` breaking backwards compatibility. That makes me reconsider depending on `cardano-api` in this package - perhaps we can move to depending only on `cardano-ledger` in the future.

I won't hold this up (hence the approval) - I'll make it work on the hydra side.

**Comment by sjoerdvisscher** [comment]
The reason `cardano-api` is breaking backwards compatibility is because `cardano-ledger` is even worse!

**Comment by sjoerdvisscher** [comment]
In fact, I only update the upper bounds of the cardano-ledger packages, but I suspect in most cases the lower bounds don't work anymore, but it's really hard to check.

**Comment by MaximilianAlgehed** [comment]
I mean, we do already depend on `cardano-ledger` here because `cardano-api`s abstractions aren't complete. But I guess the ambition is to make `cardano-api` more complete?

**Comment by MaximilianAlgehed** [comment]
@sjoerdvisscher if you bring this up to date we should be able to merge it

**Comment by sjoerdvisscher** [comment]
@MaximilianAlgehed Ah good! But first there's the issue of me being removed from the input-output-hk GitHub group, so I no longer have rights to push to this branch. Hopefully that will be resolved soon.

**Pull request closed without merging**